### PR TITLE
1653/feat/tracking recommendation

### PR DESCRIPTION
**Pull request by y-golde**
Schema : https://github.com/CoronaTeamMOD/CoronaISchemas/pull/144

Added Tracking recommendation at the bottom of the last page - controlled by redux (investigation)

### CODE REVIEW

File: client/src/components/App/Content/InvestigationForm/TrackingRecommendation/TrackingRecommendationForm.tsx [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+import AlphanumericTextField from 'commons/NoContextElements/AlphanumericTextField';
+import { setTrackingRecommendation } from 'redux/Investigation/investigationActionCreators';
+
+import useStyles from './trackingRecommendationStyles';
+import { Option } from './trackingRecommendationTypes';
+import UseTrackingRecommendationForm from './useTrackingRecommendationForm';
+
+const otherSubReason = 99;
+const defaultTrackingReason = 0;
+const trackingOptions = [
```
Comment (Noam6421) [inline client/src/components/App/Content/InvestigationForm/TrackingRecommendation/TrackingRecommendationForm.tsx:16]: maybe create a file in models for that

File: client/src/components/App/Content/InvestigationForm/TrackingRecommendation/TrackingRecommendationForm.tsx [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+            </Collapse>
+        </Grid>
+    )
+}
+
+interface Props {
+    
+}
+
+export default TrackingRecommendationForm
```
Comment (Noam6421) [inline client/src/components/App/Content/InvestigationForm/TrackingRecommendation/TrackingRecommendationForm.tsx:114]: ; at end

File: client/src/components/App/Content/InvestigationForm/TrackingRecommendation/useTrackingRecommendationForm.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import axios from 'axios';
+
+import logger from 'logger/logger';
+import { Severity } from 'models/Logger';
+
+const UseTrackingRecommendationForm = (props: Props) => {
+    const fetchSubReasonsByReason = async (reasonId : number) => {
```
Comment (Noam6421) [inline client/src/components/App/Content/InvestigationForm/TrackingRecommendation/useTrackingRecommendationForm.ts:8]: add empty line

File: client/src/components/App/Content/InvestigationForm/TrackingRecommendation/useTrackingRecommendationForm.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    return {
+        fetchSubReasonsByReason
+    }
+}
+    
+interface Props {
+        
+}
+    
+export default UseTrackingRecommendationForm
```
Comment (Noam6421) [inline client/src/components/App/Content/InvestigationForm/TrackingRecommendation/useTrackingRecommendationForm.ts:30]: ; at end

**Review by Noam6421 — CHANGES_REQUESTED** [review]

File: client/src/components/App/Content/InvestigationForm/TrackingRecommendation/useTrackingRecommendationForm.ts (hunk 4 above)
Comment (y-golde) [reply client/src/components/App/Content/InvestigationForm/TrackingRecommendation/useTrackingRecommendationForm.ts]: fixed [here](https://github.com/CoronaTeamMOD/CoronaInvestigation/pull/1426/commits/f0664bdc73f6c034c3e749827412aeef075b6225)

File: client/src/components/App/Content/InvestigationForm/TrackingRecommendation/useTrackingRecommendationForm.ts (hunk 3 above)
Comment (y-golde) [reply client/src/components/App/Content/InvestigationForm/TrackingRecommendation/useTrackingRecommendationForm.ts:9]: fixed [here](https://github.com/CoronaTeamMOD/CoronaInvestigation/pull/1426/commits/f0664bdc73f6c034c3e749827412aeef075b6225)

File: client/src/components/App/Content/InvestigationForm/TrackingRecommendation/TrackingRecommendationForm.tsx (hunk 2 above)
Comment (y-golde) [reply client/src/components/App/Content/InvestigationForm/TrackingRecommendation/TrackingRecommendationForm.tsx]: fixed [here](https://github.com/CoronaTeamMOD/CoronaInvestigation/pull/1426/commits/f0664bdc73f6c034c3e749827412aeef075b6225)

File: client/src/components/App/Content/InvestigationForm/TrackingRecommendation/TrackingRecommendationForm.tsx (hunk 1 above)
Comment (y-golde) [reply client/src/components/App/Content/InvestigationForm/TrackingRecommendation/TrackingRecommendationForm.tsx]: fixed [here](https://github.com/CoronaTeamMOD/CoronaInvestigation/pull/1426/commits/f0664bdc73f6c034c3e749827412aeef075b6225)

File: client/src/Utils/TrackingRecommendation/updateTrackingRecommendation.ts [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    const updateTrackingReccomentaion = async () => {
+        const updateInvestigationTrackingLogger = logger.setup('Update investigation tracking recommendation');
+        updateInvestigationTrackingLogger.info('launching investigation tracking recommendation request' , Severity.LOW);
+        await axios.post('/investigationInfo/updateTrackingRecommendation' , trackingRecommendation)
+            .then(() => {
+                updateInvestigationTrackingLogger.info('update tracking recommendation was successful' , Severity.LOW)
+            })
+            .catch((error) => {
+                updateInvestigationTrackingLogger.error(`got errors in server result: ${error}`, Severity.HIGH);
+                alertError('לא היה ניתן לעדכן את סיבת האיכון');
```
[the remark this answers is not in the document]
Comment (y-golde) [reply client/src/Utils/TrackingRecommendation/updateTrackingRecommendation.ts]: fixed [here](https://github.com/CoronaTeamMOD/CoronaInvestigation/pull/1426/commits/f0664bdc73f6c034c3e749827412aeef075b6225)

**Review by Noam6421 — DISMISSED** [review]

**Review by Noam6421 — APPROVED** [review]

**Review by chenFriedman — APPROVED** [review]

**Pull request merged**